packages/store/addon/-private/system/core-store.ts

```diff
@@ -71,15 +67,18 @@ import { RequestPromise } from './request-cache';
 import { PromiseProxy } from '../ts-interfaces/promise-proxies';
 import { DSModel } from '../ts-interfaces/ds-model';
 import NotificationManager from './record-notification-manager';
-import { RelationshipsSchema, AttributeSchema, AttributesSchema } from '../ts-interfaces/record-data-schemas';
+import { AttributesSchema } from '../ts-interfaces/record-data-schemas';
 import { SchemaDefinitionService } from '../ts-interfaces/schema-definition-service';
 import ShimModelClass from './model/shim-model-class';
 import RecordDataRecordWrapper from '../ts-interfaces/record-data-record-wrapper';
-import Reference from './references/reference';
+import RecordData from '../ts-interfaces/record-data';
 import { Dict } from '../ts-interfaces/utils';
 
 import constructResource from '../utils/construct-resource';
 import { errorsArrayToHash } from './errors-utils';
+
+type Relationship = import('@ember-data/record-data/-private').Relationship;
```

Comment:
My guess is you're using this as a way of working around the fact that you want *only* a type imported and *not* the runtime class, presumably to avoid circular dependencies in the module graph?

For what it's worth, the Babel build *should* handle type-only import usage correctly (as of a relatively recent fix).

```suggestion
import { Relationship } from '@ember-data/record-data/-private';
```

At least in my copy of this PR, there's no build or test issue with using the normal import here!

Reply:
Making this change causes builds to throw maximum call-stack exceeded errors, and as far as we've seen babel / babel-typescript does not strip type-only imports. We may not have gotten whatever goodness you refer to yet, or it needs a config we don't have, or it just doesn't work here.